x-pack/legacy/plugins/watcher/public/app.js

```diff
@@ -16,42 +16,35 @@ import { BASE_PATH } from './constants';
 import { LICENSE_STATUS_VALID } from '../../../common/constants';
 import { EuiCallOut, EuiLink } from '@elastic/eui';
 
-class ShareRouter extends Component {
-  static contextTypes = {
-    router: PropTypes.shape({
-      history: PropTypes.shape({
-        push: PropTypes.func.isRequired,
-        createHref: PropTypes.func.isRequired
-      }).isRequired
-    }).isRequired
-  }
-  constructor(...args) {
-    super(...args);
-    this.registerRouter();
-  }
+const ShareRouterComponent = React.memo(({ children, history }) => {
```

Comment:
CC @jloleysens Just giving you a head's up about the changes being made to this file because I know you've been working on it too. Any chance there might be merge conflicts?

Reply:
I think there will be, but they should be minor :)

Reply:
for sake of consistancy changed it back to the `Component`